Beta History: handle bulk operation errors

davelopez opened the pull request:
- Fixes `Processing operation...` message getting stuck when the history is not expected to change (due to errors during the operation, etc.)
- Includes a bunch of tests for `SelectionOperations` component
- Display an error dialog with the error details if any of the items fail to be processed
  
  ![error-dialog](https://user-images.githubusercontent.com/46503462/168333650-94370329-89bc-4032-bd6d-72e218756724.gif)


## How to test the changes?
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

davelopez committed:
Handle bulk operation errors

If the operation does not return at least one success in the `success_count`
we shouldn't expect any history changes and the 'processing' indicator
should be stopped.

The operation can also fail in different ways:
- The request fails completely (no response data).
- The `errors` in the response:
    - All items failed (success_count == 0)
    - Some items failed and some succeeded

davelopez committed:
Add jest tests for SelectionOperations component

guerler commented on client/src/components/History/CurrentHistory/HistoryOperations/SelectionOperations.vue:
```diff
@@ -10,21 +10,21 @@
             </template>
             <b-dropdown-text>
-                <span v-localize>With {{ numSelected }} selected...</span>
+                <span v-localize data-test-id="selected-count">With {{ numSelected }} selected...</span>
```
This is amazing, thank you for the fix and particularly the tests, this is **exactly** what the new history needs! I have a minor suggestion here, I suggest we use `data-description` instead of `data-test-id` and replace the dash with a space.

davelopez replied:
Done in [f4251e2](https://github.com/galaxyproject/galaxy/pull/13911/commits/f4251e2fbbec623b2abaaa775fbdaa42d4a9c913)

davelopez committed:
Replace `data-test-id` with `data-description`

davelopez committed:
Add operation error dialog

Will be displayed when some (or all) of the items in the operation could not be successfully processed.

bgruening reviewed (approved):
Thanks @davelopez
